dev/com.ibm.ws.jca.1.7_fat/fat/src/com/ibm/ws/jca/fat/FATSuite.java

```diff
@@ -16,16 +16,31 @@
 import org.junit.runners.Suite.SuiteClasses;
 
 import componenttest.rules.repeater.EmptyAction;
-import componenttest.rules.repeater.FeatureReplacementAction;
+import componenttest.rules.repeater.JakartaEE10Action;
+import componenttest.rules.repeater.JakartaEE9Action;
 import componenttest.rules.repeater.RepeatTests;
+import componenttest.topology.impl.JavaInfo;
 
 @RunWith(Suite.class)
 @SuiteClasses(JCA17Test.class)
 public class FATSuite {
-    /*
-     * EE7 will run with full fat only. EE9 will be run with lite and full fat.
-     */
     @ClassRule
-    public static RepeatTests r = RepeatTests.with(new EmptyAction().fullFATOnly())
-                    .andWith(FeatureReplacementAction.EE9_FEATURES());
+    public static RepeatTests repeat;
+
+    static {
+        // EE10 requires Java 11.  If we only specify EE10 for lite mode it will cause no tests to run which causes an error.
+        // If we are running on Java 8 have EE9 be the lite mode test to run.
+        if (JavaInfo.JAVA_VERSION >= 11) {
+            repeat = RepeatTests.with(new EmptyAction().fullFATOnly())
+                            // need widen option to handle jar file within a jar file.
+                            .andWith(new JakartaEE9Action().fullFATOnly().withWiden())
```

Comment:
For this particular one, the test is putting a jar within a rar https://github.com/OpenLiberty/open-liberty/blob/integration/dev/com.ibm.ws.jca.1.7_fat/fat/src/com/ibm/ws/jca/fat/JCA17Test.java#L65 so thought it might be needed.

Reply:
Yup, totally understandable. 
For future reference here is the transformer code that relates to this: https://github.com/eclipse/transformer/blob/0e8245c431112b52e6fb7a03a87762632d545a5c/org.eclipse.transformer/src/main/java/org/eclipse/transformer/Transformer.java#L1158-L1160

war in ear, and jar in rar are expected cases based on Jakarta EE so the transformer handles them by default.  Jar in jar is a fringe case that we support.